sdk/internal/runtime/frame_error.go

```diff
@@ -0,0 +1,43 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"fmt"
+	"runtime"
+)
+
+// NewFrameError wraps the specified error with an error that provides stack frame information.
+// Call this at the inner error's origin to provide file name and line number info with the error.
+// DO NOT ARBITRARILY CALL THIS TO WRAP ERRORS!  There MUST be only ONE error of this type in the chain.
```

Comment:
So only use this with the top-most error we return? How do we guarantee that?

Reply:
Unfortunately I don't think we can really guarantee it, outside of walking the error chain and verifying that there's only one frameError in the chain, which would also mean we have to export this type.  Maybe we want to do this in tests which means the type would need to be exported (it's still in an internal package so only we can access it).

Reply:
Hmm I see, well we can think about exporting it, but I think it's better in the internal package for now.

Reply:
It would remain in the exported from the internal package.  Exporting it would allow tests to walk the error chain looking for multiple `*FrameError` types.